tests/test_extract.py

```diff
@@ -38,10 +31,8 @@ def setUp(self):
         )
         self.configuration = config.Configuration(root_dir=MOCK_DJANGO_APP_DIR)
 
-        if not SETUP_HAS_RUN:
```

Comment:
Could you elaborate on why setup is being run multiple times now? I'd think anything that needs to be run more than once should exist somewhere other than setup, as setup is designed to run once before all the tests start.

Reply:
Thank you for asking @brian-smith-tcril . I mentioned the following in the commit:
```
Tests for `extract` script were skipped in the old days because they were part of `edx-platform` tests.
So now since we have this code in a separate package; it is more convenient to reactivate
these tests and eliminate the run-once restriction
```
for the same reason, tests were built to take less time (before they were skipped later). The approach of using a global variable to control the flow is terrible in modern code for many reasons:

* For class setup we should use `setUpClass` that will run only once instead of using a global variable like this. This is a very old code before `setUpClass` was introduced in Python 2.7
* The setup uses `_setup_i18n_test_config` which will [revert files](https://github.com/openedx/i18n-tools/blob/121700404d20a692326c8c1df0a2707f38117b61/tests/__init__.py#L24) after **each test**, not after all tests. So the first test to run will include changes of the `extract` command, while all other tests will not. Tests were passing because the [source files](https://github.com/openedx/i18n-tools/tree/723a0fbc3158afe9d0c42f4d5ed2cc6315ce480c/tests/data/mock-django-app/locale/mock/LC_MESSAGES) before doing `extract` are already formatted correctly. Meaning: they were not testing the correct files anyway except for the first test to run
* Because of the second point: my new test will fail if another test runs before it. Simply because files will be reverted and the metadata will be found in the file

So, we can edit the source files and keep the run-once mechanism, but that will not fix the fact that tests are not testing the correct files (as described in the second point above)